In [53]:
import matplotlib.pyplot as plt
import numpy as np
import math
from scipy.optimize import fsolve

a1 = 0
d = 0
def Aa(n):
    '''Succeción aritmetica'''
    return a1 + (n - 1) * d

r = 0
def Ag(n):
    '''Succeción geometrica'''
    return a1 * (r**(n - 1))

x = np.linspace(1,10,1)

# Problema 1

Considere la sucesión $a_{n} = 3n^{2} + 7$. Por medio de un código en *Python* despliega en pantalla:

1. Los primeros $5$ términos.
2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

In [54]:
def problema1_A(n):
    return (3 * (n ** 2)) + 7

### 1. Los primeros $5$ términos.

In [55]:
resultados = [problema1_A(n) for n in range(1,6)]
print(resultados)

[10, 19, 34, 55, 82]


1. termino = 10
2. termino = 19
3. termino = 34
4. termino = 55
5. termino = 82

### 2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

In [56]:
resultados = [problema1_A(n) for n in range(16,21)]
print(resultados)

[775, 874, 979, 1090, 1207]


16. termino = 775
17. termino = 874
18. termino = 979
19. termino = 1090
20. termino = 1207

# Problema 2

Considere la sucesión $g_{n} = 5n^{3}$. Implemente un código en *Python* que muestre en pantalla:

1. Los primeros $4$ términos de la sucesión.
2. Los $4$ términos que vienen inmediatamente después del octavo término.
3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

In [57]:
def problema2_G(n):
    return 5 * (n ** 3)

### 1. Los primeros $4$ términos de la sucesión.

In [58]:
resultados = [problema2_G(n) for n in range(1,5)]
print(resultados)

[5, 40, 135, 320]


1. termino = 5
2. termino = 40
3. termino = 135
4. termino = 320

### 2. Los $4$ términos que vienen inmediatamente después del octavo término.

In [59]:
resultados = [problema2_G(n) for n in range(9,13)]
print(resultados)

[3645, 5000, 6655, 8640]


9. termino = 3645
10. termino = 5000
11. termino = 6655
12. termino = 8640

### 3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

In [79]:
def problema2_buscar_(n):
    return problema2_G(n) - 40000
solucion = fsolve(problema2_buscar_,x)
print(solucion,problema2_G(20))

[20.] 40000


El termino = 40.000 si pertenece a la sucesión, en la posición 20.

# Problema 3

La sucesión de Fibonacci es una serie de números en la que cada número, desde el tercero en adelante, es la suma de los dos anteriores. Comienza con el $0$ y $1$, tal como se observa a continuación:

$$0, 1, 1, 2, 3, 5, 8, 13, ...$$

Formalmente, la sucesión se define de la siguiente manera:

\begin{split}
F(0) &= 0\\
F(1) &= 1\\
F(n) &= F(n - 1) + F(n - 2), \ \text{ para $n \geq 2$}
\end{split}

Esta sucesión se utiliza en una variedad de contextos en informática y algoritmos debido a sus propiedades matemáticas y patrones.

1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci.

### 1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.

### 2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci.

# Problema 4

Implementa un código que permita guardar en una lista la sucesión de los primeros $n$ números pares, con $n \geq 1$. Luego, determina:

1. ¿Cuál es el valor del décimo término?
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $100$ números pares.
3. ¿En qué posición se encuentra el número $58$ dentro de la sucesión?

# Problema 5

Un nuevo software tiene $500$ usuarios y se espera que el número de usuarios crezca un $15\%$ respecto al mes anterior. Considere que $b_{n}$ es el número de usuarios transcurridos $n$ meses.

1. Calcula cuántos usuarios tendrá el software al cabo de tres meses. ¿Y al cabo de seis meses?
2. Mediante un código en *Pyhton* muestra la cantidad de usuarios que tendrá el software cada mes durante el primer año.
2. ¿Cuántos usuarios tendrá en total el software luego de un año? Utilice ciclos *for* para calcular.

# Problema 6

Un estudiante de ingeniería informática planea ahorrar para un proyecto personal. Para esto, el primer mes deposita $\$12.000$ y cada mes aumenta el depósito en $\$2.000$. Considere que $a_{i}$ corresponde al depósito mensual del estudiante. Determine:

1. Implementa un código que permita mostrar en pantalla lo que depositó mensualmente durante el primer año.
2. ¿Cuánto depositó el estudiante en febrero del segundo año?
3. ¿Cuál será el total ahorrado al finalizar dos años?

In [61]:
a1 = 12000
d = 2000

resultados = [Aa(n) for n in range(25)]

print(resultados[14])
print(sum(resultados))

38000
850000


# Problema 7

Por lo general, los proyectos son organizados en varias etapas que deben completarse secuencialmente.

La fase inicial de un proyecto de software tiene una duración de $3$ meses y cada fase subsecuente reduce su duración en un $10 \%$ debido a la experiencia ganada. Determine:

1. ¿Cuánto dura la quinta fase?
2. Calcula la duración total del proyecto después de $10$ fases.

In [62]:
a1 = 3
d = -0.1
resultados = [Aa(n) for n in range(11)]
#A)
print(resultados[5])
#B)
print(sum(resultados))

2.6
28.6


# Problema 8

En una sucesión aritmética, la diferencia entre un término y el anterior es $4$ y el décimo término es $-20$. Determine:

1. El primer término de la sucesión.
2. La expresión del término de lugar $n$.
3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.
4. Determinar la posición del número $1.680$ en la secuencia. Calcule utilizando un ciclo *while*.

### 1. Encontrar el primer termino de la sucesión

In [63]:
def A1(x):
    return (x + (10 - 1) * d) - (-20)
d = 4
a1 = round(np.unique(fsolve(A1,x))[0])
print(a1)

-56


El primer termino de la suceción es -56 = A1

### 2. La expresión del término de lugar n.
A $n$ = -56 +($n$ - 1)* 4

### 3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.

In [64]:
resultados = [Aa(n) for n in range(436)]
print(resultados[100])

340


El termino en la posición 100 es 340.

### 4. Determinar la posición del número $1.680$ en la secuencia.

In [65]:
def Ax(x):
    return Aa(x) - 1680
solucion = round(np.unique(fsolve(Ax,x))[0])
print(solucion)
print(resultados[solucion])

435
1680


La posición donde An = 1680 es la posición 435.

# Problema 9

El tercer término de una sucesión geométrica es $5$ y el sexto término es $40$. Determine:

1. La razón geométrica de la sucesión.
2. El primer término de la sucesión.
3. La expresión del término de lugar $n$.
4. Utilice un código en *Python* para calcular el término que ocupa el lugar $8$.
5. ¿Qué lugar ocupa en esta sucesión el término $20.480$? Calcule utilizando un ciclo *while*.

### 1. La razón geométrica de la sucesión.

In [66]:
r = 2

La razón geométrica es 2.

### 2. El primer término de la sucesión.

In [67]:
def A1(x):
    return (x * (r**(3 - 1))) - 5
a1 = np.unique(fsolve(A1,x))[0]
print(a1)


1.25


El primer termino de la sucesión es 1.25.

### 3. La expresión del término de lugar $n$.

La expresión es $A_{n} = 1.25(2^{n-1})$

### 4. Utilice un código en *Python* para calcular el término que ocupa el lugar $8$.

In [68]:
print(Ag(8))

160.0


El término de la octava posición es 160

### 5. ¿Qué lugar ocupa en esta sucesión el término $20.480$?

# Problema 10

Una empresa de tecnología está optimizando sus servidores. Para ello, mide cada mes las transacciones por segundo (TPS) de sus servidores. Se ha obtenido que el segundo mes, el rendimiento del servidor es $1,2$ miles de TPS y el quinto mes, el rendimiento es de $2,7$ miles de TPS. Determina utilizando un código en *Python*:

1. ¿Cuál es el rendimiento el noveno mes?
2. ¿En qué mes el rendimiento es de $23,4$ miles de TPS?

Utilice dos decimales para el cálculo.

# Problema 11

Una aplicación de redes sociales inició el $1$ de enero y comenzó a ganar usuarios rápidamente. El $3$ de enero tenía $500$ usuarios y el $10$ de enero tenía $1.501$ usuarios. Si el número de usuarios sigue creciendo de manera constante, determine:

1. ¿Con cuántos usuarios comenzó a funcionar la aplicación?
2. ¿Cuántos usuarios tendrá la aplicación el $31$ de enero?
3. Si la aplicación tiene $5.505$ usuarios, ¿cuántos meses lleva funcionando?



### Encontrar la $d$.

In [69]:
def encontrar_d(d):
    return (2 * d - 500) - (9 * d - 1501)
d = round(fsolve(encontrar_d,1)[0])
print(d)

143


### 1. ¿Con cuántos usuarios comenzó a funcionar la aplicación?

In [70]:
def A1(x):
    return (x + (3 - 1) * d) - 500
a1 = round(np.unique(fsolve(A1,x))[0])
print(a1)

214


La aplicación empezo con 214 usuarios.

### 2. ¿Cuántos usuarios tendrá la aplicación el $31$ de enero?

In [71]:
print(Aa(31))

4504


La aplicación tendria 4504 usuarios el día 31 de enero.

### 3. Si la aplicación tiene $5.505$ usuarios, ¿cuántos meses lleva funcionando?

In [72]:
def Ax(x):
    return Aa(x) - 5505
solucion = round(np.unique(fsolve(Ax,x))[0])
print(solucion)
print(Aa(38))
print(round(1+7/31,2))

38
5505
1.23


Si la aplicación tiene 5505, deberian pasar 1.23 meses o más exactamente 38 dias.

# Problema 12

Una empresa de publicidad ofrece para un puesto de trabajo un salario inicial y un aumento mensual en una razón constante. Javier se presenta a la entrevista, y el encargado del departamento de recursos humanos le explica que en el segundo mes de trabajo su sueldo será de $\$402.000$, y que en el tercer mes de trabajo será de $\$404.010$. Determine:

1. El salario aproximado que Javier recibiría en el décimo mes de trabajo en la empresa.
2. ¿Cuál sería el ingreso total recibido por los sueldos de Javier durante el tercer año de trabajo?

### 1. El salario aproximado que Javier recibiría en el décimo mes de trabajo en la empresa.

### 2. ¿Cuál sería el ingreso total recibido por los sueldos de Javier durante el tercer año de trabajo?

# Problema 13

Una empresa de ciberseguridad capacita trabajadores cada trimestre. En el primer trimestre capacitaron a $30$ trabajadores, en el segundo trimestre a $60$ trabajadores, y en el tercer trimestre a $120$ trabajadores. Si en los siguientes trimestres la empresa mantiene el mismo patrón de crecimiento, determine utilizando *Python*:

1. ¿A cuántos trabajadores capacitarán en el octavo trimestre?
2. Si la empresa lleva en el mercado $3$ años, ¿a cuántos trabajadores han capacitado en total?

### 1. ¿A cuántos trabajadores capacitarán en el octavo trimestre?

In [73]:
r = 2
a1 = 30
print(Ag(8))

3840


### 2. Si la empresa lleva en el mercado $3$ años, ¿a cuántos trabajadores han capacitado en total?

In [74]:
resultados = [Ag(n) for n in range(1,13)]
print(sum(resultados))

122850


# Problema 14

Al llenar un tambor con agua, se registra que durante el minuto $4$ el tambor recibe $128$ litros, y durante el minuto $9$ recibe $88$ litros. Sabiendo que en cada minuto recibe una cantidad $d$ de litros de agua menos que en el minuto anterior, determine:

1. ¿En qué instante el tambor deja de recibir agua?
2. ¿Cuántos litros recibe en total el tambor?

Realice los cálculos anteriores utilizando *Python*.

In [75]:
def encontrar_d(d):
    return (3*d - 128) - (8*d - 88)
d = fsolve(encontrar_d,1)
def A1(x):
    return (x + (4 - 1) * d) - 128
a1 = round(np.unique(fsolve(A1,x))[0])
print(f"A1 = {a1}\nd = {d}")

A1 = 152
d = [-8.]


### 1. ¿En qué instante el tambor deja de recibir agua?

In [76]:
solucion = fsolve(Aa,x)
print(solucion,Aa(20),Aa(21))

[20.] [5.68434189e-14] [-8.]


El tambor dejaria de recibir agua en el minuto 20.

### 2. ¿Cuántos litros recibe en total el tambor?

In [77]:
resultados = [Aa(n) for n in range(1,21)]
print(sum(resultados))

[1520.]


El tambor recibe en total 1520 litros de agua.